File: 7-kyu/alphabet-symmetry/c++/solution.cpp
#include <cctype>
#include <vector>
#include <string>

std::vector<int> solve(std::vector<std::string> arr)
{  
    std::vector<int> ret;
    for (const auto &word : arr) {
        unsigned chars_matching_pos = 0;
        for (unsigned i = 0; i < word.size(); i++) {
            char ch = std::tolower(word[i]);
            if (i == ch - 'a') chars_matching_pos++;
        }
        ret.push_back(chars_matching_pos);
    }
    return ret;
}

File: 7-kyu/alphabet-symmetry/javascript/solution.js
const solve = arr => {  
  let ret = [];
  for (const word of arr) {
    let charsMatching = 0;
    for (let i = 0; i < word.length; i++) {
      const wordLower = word.toLowerCase();
      if (wordLower.charCodeAt(i) - 97 == i) charsMatching++; // charCode of 'a' == 97
    }
    ret.push(charsMatching);
  }
  return ret;
};
